Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -491,6 +506,134 @@ NWC24::ErrorCode NetKDRequestDevice::DetermineSubtask(u16 entry_index,
+  });
+
+  {
+    std::lock_guard lg(m_scheduler_buffer_lock);
+    m_scheduler_buffer[4] = Common::swap32(static_cast<u32>(CurrentFunction::Send));
+  }
+
+  m_send_list.ReadSendList();
+  const std::string auth =
+      fmt::format("mlid=w{}\r\npasswd={}", m_config.Id(), m_config.GetPassword());
+  std::vector<Common::HttpRequest::Multiform> multiform = {{"mlid", auth}};
+
+  std::vector<u32> mails = m_send_list.GetMailToSend();
+  for (const u32 file_index : mails)
+  {
+    const u32 entry_id = m_send_list.GetEntryId(file_index);
+    const u32 mail_size = m_send_list.GetMailSize(file_index);
+    if (mail_size > MAX_MAIL_SIZE)
+    {
+      WARN_LOG_FMT(IOS_WC24,
+                   "NET_KD_REQ: IOCTL_NWC24_SEND_MAIL_NOW: Mail at index {} was too large to send.",
+                   entry_id);
+      LogError(ErrorType::SendMail, NWC24::WC24_MSG_TOO_BIG);
+
+      NWC24::ErrorCode res = m_send_list.DeleteMessage(file_index);
+      if (res != NWC24::WC24_OK)
+      {
+        LogError(ErrorType::SendMail, res);
+      }
+      mails.erase(std::remove(mails.begin(), mails.end(), file_index), mails.end());
+      continue;
+    }
+
+    std::vector<u8> mail_data(mail_size);
+    NWC24::ErrorCode res = NWC24::ReadFromVFF(
+        NWC24::Mail::SEND_BOX_PATH, m_send_list.GetMailPath(file_index), m_ios.GetFS(), mail_data);
+    if (res != NWC24::WC24_OK)
+    {
+      ERROR_LOG_FMT(IOS_WC24, "Reading mail at index {} failed with error code {}.", entry_id,
+                    static_cast<s32>(res));
+      LogError(ErrorType::SendMail, NWC24::WC24_MSG_DAMAGED);
+      res = m_send_list.DeleteMessage(file_index);
+      if (res != NWC24::WC24_OK)
+      {
+        LogError(ErrorType::SendMail, res);
+      }
+
+      mails.erase(std::remove(mails.begin(), mails.end(), file_index), mails.end());
+      continue;
+    }
+
+    const std::string mail_str = {mail_data.begin(), mail_data.end()};
+
+    multiform.push_back({fmt::format("m{}", entry_id), mail_str});
+  }
+
+  const Common::HttpRequest::Response response =
+      m_http.PostMultiform(m_config.GetSendURL(), multiform);
+
+  if (!response)
+  {
+    ERROR_LOG_FMT(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_SEND_MAIL_NOW: Failed to request data at {}.",
+                  m_config.GetSendURL());
+    return NWC24::WC24_ERR_SERVER;
+  }
+
+  // Now check if any mail failed to save to the server.
+  const std::string response_str = {response->begin(), response->end()};
+  const std::string code = GetValueFromCGIResponse(response_str, "cd");
+  if (code != "100")
+  {
+    ERROR_LOG_FMT(
+        IOS_WC24,
+        "NET_KD_REQ: IOCTL_NWC24_CHECK_MAIL_NOW: Mail server returned non-success code: {}", code);
+    return NWC24::WC24_ERR_SERVER;
+  }
+
+  // Reverse in order to delete from bottom to top of the send list.
+  // We do this to ensure that new entries can be written as close to the beginning of the file as
+  // possible.
+  std::reverse(mails.begin(), mails.end());
```

Comment:
Would it not be possible to use `.rbegin()` and `.rend()` to get a reverse view of the container, rather than doing an inplace reversal?